python/pyspark/ml/feature.py

```diff
@@ -2111,26 +2112,45 @@ class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol, HasHandleInvalid,
     >>> loadedInverter = IndexToString.load(indexToStringPath)
     >>> loadedInverter.getLabels() == inverter.getLabels()
     True
+    >>> stringIndexer.getStringOrderType()
+    'frequencyDesc'
+    >>> stringIndexer = StringIndexer(inputCol="label", outputCol="indexed", handleInvalid='error',
+    ...     stringOrderType="alphabetDesc")
+    >>> model = stringIndexer.fit(stringIndDf)
+    >>> td = model.transform(stringIndDf)
+    >>> sorted(set([(i[0], i[1]) for i in td.select(td.id, td.indexed).collect()]),
+    ...     key=lambda x: x[0])
+    [(0, 2.0), (1, 1.0), (2, 0.0), (3, 2.0), (4, 2.0), (5, 0.0)]
 
     .. versionadded:: 1.4.0
     """
 
+    stringOrderType = Param(Params._dummy(), "stringOrderType",
+                            "How to order labels of string column. The first label after " +
+                            "ordering is assigned an index of 0. Supported options: " +
+                            "frequencyDesc, frequencyAsc, alphabetDesc, alphabetAsc.",
```

Comment:
@felixcheung `stringOrderType` is not a shared trait on the Scala side, and I thought only the shared traits should be automatically generated. 
I have looked at the code for other ML transformers, and many of them hard coded, for example, [Imputer](https://github.com/apache/spark/blob/master/python/pyspark/ml/feature.py#L933) and [OneHotEncoder](https://github.com/apache/spark/blob/master/python/pyspark/ml/feature.py#L1520). 
Please let me know if I'm wrong, and a reference to example would be greatly appreciated. Thanks much!